frontends/web/src/components/transactions/components/details.tsx

```diff
@@ -0,0 +1,190 @@
+import { useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as accountApi from '../../../api/account';
+import { A } from '../../anchor/anchor';
+import { Dialog } from '../../dialog/dialog';
+import { FiatConversion } from '../../rates/rates';
+import { Amount } from '../../../components/amount/amount';
+import { Note } from '../note';
+import { TxDetail } from './detail';
+import { Arrow } from './arrow';
+import { TxDate } from './date';
+import { TxStatus } from './status';
+import { AddressOrTxID } from './address-or-txid';
+import parentStyle from '../transaction.module.css';
+
+type TProps = {
+  open: boolean;
+  onClose: () => void;
+  accountCode: string;
+  internalID: string;
+  note: string;
+  status: accountApi.TTxStatus;
+  type: accountApi.TTxType;
+  numConfirmations: number;
+  numConfirmationsComplete: number;
+  time: string | null;
+  amount: accountApi.IAmount;
+  sign: string;
+  typeClassName: string;
+  feeRatePerKb: accountApi.IAmount;
+  explorerURL: string;
+}
+
+export const TxDetailsDialog = ({
+  open,
+  onClose,
+  accountCode,
+  internalID,
+  note,
+  status,
+  type,
+  numConfirmations,
+  numConfirmationsComplete,
+  time,
+  amount,
+  sign,
+  typeClassName,
+  feeRatePerKb,
+  explorerURL,
+}: TProps) => {
+  const { t } = useTranslation();
+
+  const transactionInfo = useRef<accountApi.ITransaction | null>();
+
+  useEffect(() => {
+    if (!transactionInfo.current) {
+      accountApi.getTransaction(accountCode, internalID).then(transaction => {
+        if (!transaction) {
+          console.error('Unable to retrieve transaction ' + internalID);
+        }
+        transactionInfo.current = transaction;
+      }).catch(console.error);
+    }
+  }, [accountCode, internalID]);
+
+  // Amount and Confirmations info are displayed using props data
+  // instead of transactionInfo because they are live updated.
+  return (
+    <Dialog
+      open={open && !!transactionInfo.current}
+      title={t('transaction.details.title')}
+      onClose={onClose}
+      slim
+      medium>
+      {transactionInfo.current && (
+        <>
+          <Note
+            accountCode={accountCode}
+            internalID={internalID}
+            note={note}
+          />
+          <TxDetail label={t('transaction.details.type')}>
+            <Arrow
+              status={status}
+              type={type}
+            />
+          </TxDetail>
+          <TxDetail label={t('transaction.confirmation')}>{numConfirmations}</TxDetail>
+          <TxStatus
+            status={status}
+            numConfirmations={numConfirmations}
+            numConfirmationsComplete={numConfirmationsComplete}
+            detail
+          />
+          <TxDate time={time} detail />
```

Comment:
Instead of 1 component with 2 layouts depending on detail prob. I think you should just export 2 different components from that module without using detail prop.

Reply:
yes this makes sense, I also changed it for all other components (that you did not review yet)